.github/workflows/publish.yaml

```
@@ -24,22 +24,39 @@ jobs:
         with:
           strip_v: true
 
+  get-previous-tag:
+    runs-on: ubuntu-22.04
+    outputs:
+      tag: ${{ steps.previous-tag.outputs.tag }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Get previous tag
+        id: previous-tag
+        run: |
+          git fetch -a
+          tags=($(git tag --sort="-creatordate"))
+          previous_tag=${tags[1]}
+          echo "tag=$previous_tag" >> $GITHUB_OUTPUT
```

i think you'll need to strip the `v` here to be consistent with getting the current tag:
```suggestion
          echo "tag=${previous_tag#v}" >> $GITHUB_OUTPUT
```